lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -845,7 +851,13 @@ export abstract class CacheManager implements ICacheManager {
             this.commonLogger.info("CacheManager:getAccessToken - No token found");
             return null;
         } else if (numAccessTokens > 1) {
-            throw ClientAuthError.createMultipleMatchingTokensInCacheError();
+            this.commonLogger.info(
+                "CacheManager:getAccessToken - Multiple access tokens found, clearing them"
```

Comment:
This does not seem to cover refresh token, which I believe can also get into this state?

Reply:
AFAIK this issue affects id and access tokens only. Also, only first matched RT is returned by [getRefreshToken()](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/src/cache/CacheManager.ts#L1270)

Reply:
Previously the exception was thrown only for access and ID tokens, not refresh. The first token in the collection was always returned for all three tokens. With this fix, there could still be 2 duplicate RTs in the cache but the first one is returned. The questions are why wasn't the exception thrown for multiple RTs and should this fix also apply to RTs then too?

Reply:
We can't remove duplicated RTs as it essentially logs out user completely. I will instrument `getRefreshToken` function to check how often it comes across multiple matches to make informed decision whether it is worth making a change. Long-term goal is to implement cache schema versioning. Meanwhile, the logic stays the same.